addons/l10n_ch/report/swissqr_report.xml

```diff
@@ -44,12 +44,7 @@
                             <span class="swissqr_text content" t-field="o.company_id.city"/><br/>
 
                             <span class="swissqr_text title">Reference</span><br/>
-                            <t t-if="o.partner_bank_id.currency_id.name == 'EUR'">
-                                <span class="swissqr_text content" t-field="o.partner_bank_id.bank_id.l10n_ch_postal_eur"/><br/>
-                            </t>
-                            <t t-if="o.partner_bank_id.currency_id.name == 'CHF'">
-                                <span class="swissqr_text content" t-field="o.partner_bank_id.bank_id.l10n_ch_postal_chf"/><br/>
-                            </t>
+                            <span class="swissqr_text content" t-esc="o.reference"/><br/>
```

Comment:
Here we can do better.

In case of QR-IBAN we want to use the `l10n_ch_isr_number` field.
As QR-IBAN is the replacement for the postal account number.

Using o.reference here doesn't work because the structured ref is not copied in this field (that also could be the issue)

Reply:
> Here we can do better.
> 
> In case of QR-IBAN we want to use the `l10n_ch_isr_number` field.
> As QR-IBAN is the replacement for the postal account number.
> 
> Using o.reference here doesn't work because the structured ref is not copied in this field (that also could be the issue)

Are you sure you configured your company properly? Because it should be copied there if you did https://github.com/odoo/odoo/blob/12.0/addons/l10n_ch/models/account_invoice.py#L215


> @oco-odoo my fix here https://github.com/odoo/odoo/pull/35931/files#diff-4fbe6300e9eaf06e47eb21f54b583724R33
> was bad considering we want to be able to print QR Invoice without structured references, so it's missing a if for the QR-Bill with NON type.
> 
> But that line underline where should come from the reference we want in that "Reference" displayed on the report. So `o.reference` is not that good as we absolutely need to get the value from `l10n_ch_isr_number_spaced` field. And we do want this is the human readable version with spaces.

Using 'reference' directly allows always having the right value, even if we don't have QRR, as the decoupling between 'QRR' and 'not QRR' is done in Python, when generating the QR-code. Adding the spaces means we would either need to pass the spaced field to ref or change that behavior entirely. I don't thing we should refactor again. But, will passing the spaces be okay ? Is a QR-code generated with a QRR containing spaces ok ? Or  could simply just remove all spaces in the code generating the QR-code when treating a QRR ?What do you think of that?

Reply:
Ok I might be wrong about the reference, but I don't see which config would have fixed it. When creating my invoices I had a content in l10n_ch_isr_number but reference was empty.

In short:

There is a difference between the reference inside the QR Code and the reference printed to a human being.

:robot: QR code should contain the QRR without spaces: `210000000003139471430009017`
:woman: the report should contain the QRR with spaces: `21 00000 00003 13947 14300 09017`

You can check this in document https://www.paymentstandards.ch/dam/downloads/ig-qr-bill-en.pdf page 38 and page 39

Reply:
I must add that if it's not a valid QRR or SCOR `reference` (let's keep SCOR outside of scope for now) we shouldn't be using the reference field as a structured reference but as additional information.

Thus keeping the use of reference is good but we need to check what we push to the qweb report (printed and in the report)

Reply:
@oco-odoo going back to it we don't want the spaces in the field `reference` due to how the reconciliation works, as spaces would be considered as separator and show the structured ref as multiple small refs. (IMO this should be improved on the reconciliation side but we have no choice here)